packages/destination-actions/src/destinations/absmartly/event.ts

```diff
@@ -0,0 +1,74 @@
+import { InputField, JSONObject, ModifiedResponse, RequestClient } from '@segment/actions-core'
+import { Settings } from './generated-types'
+import { PublishRequestUnit } from './unit'
+import { PublishRequestAttribute } from './attribute'
+import { PublishRequestGoal } from './goal'
+import { Data } from 'ws'
+
+export interface PublishRequestEvent {
+  publishedAt: number
+  units: PublishRequestUnit[]
+  goals?: PublishRequestGoal[]
+  exposures?: JSONObject[]
+  attributes?: PublishRequestAttribute[]
+}
+
+export interface DefaultPayload {
+  publishedAt: string | number
+  agent?: string
+  application?: string
+}
+
+export const defaultEventFields: Record<string, InputField> = {
+  publishedAt: {
+    label: 'Event Sent Time',
+    type: 'datetime',
+    required: true,
+    description:
+      'Exact timestamp when the event was sent (measured by the client clock). Must be an ISO 8601 date-time string, or a Unix timestamp (milliseconds) number',
+    default: {
+      '@path': '$.sentAt'
+    }
+  },
+  agent: {
+    label: 'Agent',
+    type: 'string',
+    required: false,
+    description: 'Optional agent identifier that originated the event. Used to identify which SDK generated the event.',
+    default: 'segment'
```

Comment:
```suggestion
    default: {'@path': '$.context.library.name'}
```

This will pull the actual name of the Segment library which was used to send the event. e.g. analytics.js for Web